packages/core/src/tools/utils/urlPolyfill.ts

```diff
@@ -34,11 +34,14 @@ export function getHash(url: string) {
 }
 
 export function buildUrl(url: string, base?: string) {
-  if (checkURLSupported()) {
+  const supportedURL = getSupportedUrl()
+  if (supportedURL) {
     try {
-      return base !== undefined ? new URL(url, base) : new URL(url)
+      return base !== undefined ? new supportedURL(url, base) : new supportedURL(url)
     } catch (error) {
-      throw new Error(`Failed to construct URL. ${jsonStringify({ url, base })!}`)
+      throw new Error(
+        `Failed to construct URL. ${jsonStringify({ url, base, error: error instanceof Error ? error.message : '' })!}`
+      )
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** you could use `String(error)`, is simpler and handles any kind of error:

```suggestion
      throw new Error(
        `Failed to construct URL: ${String(error)} ${jsonStringify({ url, base })!}`
      )
```